tests/test_official.py

```diff
@@ -264,9 +287,222 @@ def check_defaults_type(ptb_param: inspect.Parameter) -> bool:
+        ptb_annotation = _unionizer(ptb_annotation, False)
+        print(ptb_annotation, "after unionizer")
+
+        # Last step, we need to use get_origin to get the original type, since using get_args
+        # above will strip that out.
+        wrapped = get_origin(ptb_param.annotation)
+        if wrapped is not None:
+            # collections.abc.Sequence -> typing.Sequence
+            if "collections.abc.Sequence" in str(wrapped):
+                wrapped = Sequence
+            ptb_annotation = wrapped[ptb_annotation]
+            print(wrapped, "after wrapped", ptb_annotation)
+        # We have put back our annotation together after removing the NoneType!
+
+    # Now let's do the checking, starting with "Array of ..." types.
+    if "Array of " in tg_param_type:
+        print("array of ", ptb_param.annotation)
+
+        # For exceptions just check if they contain the annotation
+        array_of_exceptions = {  # We accept more types than the official API
+            "results": "InlineQueryResult",
+            "commands": "BotCommand",
+            "keyboard": "KeyboardButton",
+        }
+        if ptb_param.name in array_of_exceptions:
+            return array_of_exceptions[ptb_param.name] in str(ptb_annotation)
+
+        # let's import obj
+        pattern = r"Array of(?: Array of)? ([\w\,\s]*)"
+        obj_match: re.Match | None = re.search(pattern, tg_param_type)  # extract obj from string
+        if obj_match is None:
+            raise AssertionError(f"Array of {tg_param_type} not found in {ptb_param.name}")
+        obj_str: str = obj_match.group(1)
+        # is obj a regular type like str?
+        array_of_mapped: set[type] = _get_params_base(obj_str, TYPE_MAPPING)
+
+        if len(array_of_mapped) == 0:  # no match found, it's from telegram module
+            print("from tg module")
+            # it could be a list of objects, so let's check that:
+            objs = _extract_words(obj_str)
+            # let's unionize all the objects, with and without ForwardRefs.
+            unionized_objs: list[type] = [_unionizer(objs, True), _unionizer(objs, False)]
+        else:
+            print("from TYPE_MAPPING")
+            unionized_objs = [array_of_mapped.pop()]
+
+        assert mapped_type is Sequence
+        # This means it is Array of Array of [obj]
+        if "Array of Array of" in tg_param_type:
+            # print('isinstance 1', Sequence[Sequence[obj]], ptb_annotation)
+            return any(Sequence[Sequence[o]] == ptb_annotation for o in unionized_objs)
+
+        # This means it is Array of [obj]
+        # print('isinstance 2', mapped_type[obj], ptb_annotation)
+        return any(mapped_type[o] == ptb_annotation for o in unionized_objs)
+
+    DEFAULT_VAL_PARAMS = {
+        "parse_mode",
+        "disable_notification",
+        "allow_sending_without_reply",
+        "protect_content",
+        "disable_web_page_preview",
+        "explanation_parse_mode",
+    }
```

Comment:
Can we directly iterate over the attributes of `Defaults` here? Avoids duplication …